### ISSUE TITLE
how to round trip Display and EnumString with default="true"

### ISSUE THREAD

**Issue by thedavidmeister** [Peternator7/strum#86]
I have some default variant that contains a string, like:

```rust
#[derive(Display, EnumString)]
FooEnum {
  #[strum(default="true")]
  Bar(String)
}
```

but if i do:

```rust
FooEnum::Bar("baz".into()).to_string();
```

i get `"Bar"` not `"baz"` so then there's no way to round trip it back into the original `FooEnum`

**Comment by Peternator7** [comment]
This is a breaking change, but it seems internally consistent so I'd be open to fixing/changing it. Is this something you're opening to doing?

**Comment by thedavidmeister** [comment]
not sure... it came up for me during a refactor of a crate where this was being used... but i'm not sure if i'll be using this crate moving forward on that project as we might not even be using the enum it was relevant to any more

**Comment by fullermd** [comment]
IWBNI this could be extended to AsRefStr.  As it currently stands, `(foo.to_string() != foo.as_ref())`, which is confusing (and fertile ground for confusion you don't notice right away).

**Comment by Peternator7** [comment]
This is implemented in the linked PR #270 

@fullermd, can you open a new issue for that request. The issue is that for `MyEnum::Variant(T)`, T doesn't have to be a string so it might implement `Display` but not `AsRef<str>`. That makes it a breaking change which might be fine to be more consisent, but it'd be nice to discuss in a new thread that isn't 3 years old
